minecode/visitors/generic.py

```diff
@@ -77,3 +76,86 @@ def process_request(purl_str):
 
     if error_msg:
         return error_msg
+
+
+def map_directory_listed_package(package_url):
+    """
+    Add a directory listed `package_url` to the PackageDB.
+
+    Return an error string if any errors are encountered during the process
+    """
+    from minecode.model_utils import add_package_to_scan_queue
+    from minecode.model_utils import merge_or_create_package
+
+    packages = [p for p in info(str(package_url)) or []]
+
+    if not packages:
+        error = f"Could not find package using fetchcode: {package_url}"
+        logger.error(error)
+        return error
+
+    package_data = packages[0].to_dict()
+
+    # Remove obsolete Package fields see https://github.com/nexB/fetchcode/issues/108
```

Comment:
Can you wrap each del block in a try/except KeyError? and move this code in a function?
Otherwise this will fail when FetchCode is updated.